netty/src/test/java/io/grpc/netty/TlsTest.java

```diff
@@ -101,17 +101,24 @@ public void setUp() throws NoSuchAlgorithmException {
         Assume.assumeTrue(OpenSsl.isAvailable());
         sslProvider = SslProvider.OPENSSL;
         break;
-      case JETTY:
+      case JDK:
         Assume.assumeTrue(Arrays.asList(
             SSLContext.getDefault().getSupportedSSLParameters().getCipherSuites())
             .contains("TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"));
         sslProvider = SslProvider.JDK;
         jdkProvider = Security.getProvider("SunJSSE");
         Assume.assumeNotNull(jdkProvider);
         try {
-          GrpcSslContexts.configure(SslContextBuilder.forClient(), jdkProvider);
-        } catch (IllegalArgumentException ex) {
-          Assume.assumeNoException("Jetty ALPN does not seem available", ex);
+          // Check for presence of an (ironic) class added in Java 9
+          Class.forName("java.lang.Runtime$Version");
```

Comment:
With grouping the Jetty and JDK test cases here, it's not immediately clear that Java 9's presence means we should skip the call to `GrpcSslContexts.configure`. It seems like the configuration code path would actually prefer Jetty (if set up) over Java 9 ALPN.

What breaks if the test just invokes `GrpcSslContexts.configure` here and tries to let that "do the right thing" on Java 9?

Reply:
So here's what's going on:
1) If on Java 9+, we want to run the test
2) When on Java 7/8, we want to run the test if Jetty is available. We are abusing `configure()` to find out if Jetty is available, which is a hack but "good enough." The return value of `configure()` is ignored
3) If the two options above aren't available, `Assume` will skip the test.

The problem is that configure() was broken for Java 9, so the assume skipped the test! My bug fix makes it so configure() no longer throws for Java 9+, so the test would begin running again. But avoiding the configure() for Java 9 guarantees that we'll be able to notice problems with configure() throwing, again.